lib/mpl_toolkits/mplot3d/art3d.py

```diff
@@ -627,37 +627,41 @@ def do_3d_projection(self, renderer):
             self.update_scalarmappable()
             self._facecolors3d = self._facecolors
 
-        txs, tys, tzs = proj3d._proj_transform_vec(self._vec, renderer.M)
-        xyzlist = [(txs[sl], tys[sl], tzs[sl]) for sl in self._segslices]
+        num_faces, num_face_vertices, _ = self._segments.shape
+        vecs = self._segments.reshape(-1, 3).T
+        _, _, pzs = pvecs = proj3d.proj_transform_vectors(vecs, renderer.M)
```

Comment:
You don't need `pzs` here
```suggestion
        pvecs = proj3d.proj_transform_vectors(vecs, renderer.M)
```

Reply:
I think that `pzs` is slightly nicer than doing all this `pvecs.shape[1] > 0` and `pvecs[3]` business later, but I can change it if you'd prefer.

Reply:
If nothing else, `pzs = pvecs[3]` instead of `_, _, pzs` saves the construction of two discarded array views.

`pvecs.shape[1] > 0` could just as easily be written `pvecs.size > 0`.

Reply:
Yes, but honestly I would rather avoid getting into micro-optimizing this code. This is totally not where the bottlenecks are right now! Saving a few microseconds makes absolutely no difference, because the rendering pipeline still takes at least a _few seconds_ to process those plots (likely because each face is represented as an individual Python object instead of a batched array).

Reply:
Fair enough. My suggestionn below makes this point moot anyway.